pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -373,7 +373,10 @@ def select_training_type_plugin(self):
         elif self.use_horovod:
             plugin = HorovodPlugin(parallel_devices=self.parallel_devices)
         elif self.on_tpu:
-            plugin = SingleTPUPlugin(self.tpu_id)
+            if isinstance(self.tpu_cores, list):
+                plugin = SingleTPUPlugin(self.tpu_id)
+            else:
+                plugin = TPUSpawnPlugin(parallel_devices=list(range(self.tpu_cores)))
```

Comment:
this was just a temporary fix. I think `TPUSpawnPlugin` is meant to be called above within `elif self.use_ddp` block.